modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java

```diff
@@ -1963,4 +1977,92 @@ private Optional<Response> updateSchedulingInformation(
+                  @RestResponse(description = "The track has been added to the event.", responseCode = HttpServletResponse.SC_OK),
+                  @RestResponse(description = "The request is invalid or inconsistent.", responseCode = HttpServletResponse.SC_BAD_REQUEST),
+          })
+  public Response updateFlavorWithTrack(@HeaderParam("Accept") String acceptHeader, @PathParam("eventId") String id,
+          @Context HttpServletRequest request) {
+    logger.debug("updateFlavorWithTrack called");
+    try {
+      boolean overwriteExisting = false;
+      MediaPackageElementFlavor tmpFlavor = MediaPackageElementFlavor.parseFlavor("addTrack/temporary");
+      MediaPackageElementFlavor newFlavor = null;
+      Opt<Event> event;
+
+      try {
+        event = indexService.getEvent(id, elasticsearchIndex);
+      } catch (SearchIndexException e) {
+        return RestUtil.R.badRequest(String.format("Error while searching for event with id %s; %s", id, e.getMessage()));
+      }
+
+      if (event.isNone()) {
+        return ApiResponses.notFound(String.format("Unable to find event with id '%s'", id));
+      }
+      MediaPackage mp = indexService.getEventMediapackage(event.get());
+
+      if (!ServletFileUpload.isMultipartContent(request)) {
+        throw new IllegalArgumentException("No multipart content");
+      }
+      for (FileItemIterator iter = new ServletFileUpload().getItemIterator(request); iter.hasNext();) {
+        FileItemStream item = iter.next();
+        String fieldName = item.getFieldName();
+        if (item.isFormField()) {
+          if ("flavor".equals(fieldName)) {
+            String flavorString = Streams.asString(item.openStream());
+            try {
+              newFlavor = MediaPackageElementFlavor.parseFlavor(flavorString);
+            } catch (IllegalArgumentException e) {
+              return RestUtil.R.badRequest(String.format("Could not parse flavor %s; %s", flavorString, e.getMessage()));
+            }
+          } else if ("overwriteExisting".equals(fieldName)) {
+            overwriteExisting = Boolean.parseBoolean(Streams.asString(item.openStream()));
+          }
+        } else {
+          // Add track with temporary flavor
+          if ("track".equals(item.getFieldName())) {
+            mp = ingestService.addTrack(item.openStream(), item.getName(), tmpFlavor, mp);
+          }
+        }
+      }
+
+      if (overwriteExisting) {
+        // remove existing attachments of the new flavor
+        Track[] existing = mp.getTracks(newFlavor);
+        for (int i = 0; i < existing.length; i++) {
+          mp.remove(existing[i]);
+          logger.debug("Overwriting existing asset {} {}", tmpFlavor, newFlavor);
+        }
+      }
+      // correct the flavor of the new attachment
+      Track[] elArray = mp.getTracks(tmpFlavor);
+      elArray[0].setFlavor(newFlavor);
+      logger.debug("Updated asset {} {}", tmpFlavor, newFlavor);
```

Comment:
It's possible to have multiple track parameters that would only correct the flavor for the first entry.

![image](https://user-images.githubusercontent.com/35195803/168346081-b14a5008-b00a-4a3e-9f68-4140be4e8fce.png)